{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"id":"bHXKi5EOIjn2","tags":[],"executionInfo":{"status":"ok","timestamp":1693919010841,"user_tz":-60,"elapsed":455,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["import numpy as np\n","import os\n","import sys"]},{"cell_type":"code","execution_count":2,"metadata":{"tags":[],"colab":{"base_uri":"https://localhost:8080/"},"id":"YXYQPh_DtWlB","executionInfo":{"status":"ok","timestamp":1693919018361,"user_tz":-60,"elapsed":7522,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"5e9c70cf-1b7d-4cab-8cb8-5bf1fd354b8e"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting tmscoring\n","  Downloading tmscoring-0.4.post0.tar.gz (5.3 kB)\n","  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","Building wheels for collected packages: tmscoring\n","  Building wheel for tmscoring (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for tmscoring: filename=tmscoring-0.4.post0-py3-none-any.whl size=5754 sha256=4fbbcf1a16f1d4b281ee63e1a2fad72331cc6fe94f5cf8414770708012fcddfe\n","  Stored in directory: /root/.cache/pip/wheels/42/7f/85/dd37516a1a64921564b4a7cb3e5640eadecb624c215ded2480\n","Successfully built tmscoring\n","Installing collected packages: tmscoring\n","Successfully installed tmscoring-0.4.post0\n"]}],"source":["!pip install tmscoring"]},{"cell_type":"code","execution_count":3,"metadata":{"tags":[],"colab":{"base_uri":"https://localhost:8080/"},"id":"xtGq0_TetWlC","executionInfo":{"status":"ok","timestamp":1693919100575,"user_tz":-60,"elapsed":82216,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"570fd9bf-4aba-4c3d-e216-1deb80972cc1"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting iminuit<2\n","  Downloading iminuit-1.5.4.tar.gz (755 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m756.0/756.0 kB\u001b[0m \u001b[31m9.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","Requirement already satisfied: numpy>=1.11.3 in /usr/local/lib/python3.10/dist-packages (from iminuit<2) (1.23.5)\n","Building wheels for collected packages: iminuit\n","  Building wheel for iminuit (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for iminuit: filename=iminuit-1.5.4-cp310-cp310-linux_x86_64.whl size=3651399 sha256=b820dc8696e6ba24e4088cedc4c258bc7f25236304b853cda53d2d43b3cc2af1\n","  Stored in directory: /root/.cache/pip/wheels/19/26/42/40159d2024e19468692b49ad455e9d3c499800c7000f49c94a\n","Successfully built iminuit\n","Installing collected packages: iminuit\n","Successfully installed iminuit-1.5.4\n"]}],"source":["!pip install \"iminuit<2\""]},{"cell_type":"code","execution_count":null,"metadata":{"tags":[],"id":"pgvE1DIXtWlC"},"outputs":[],"source":["#!pip install plotly"]},{"cell_type":"code","execution_count":4,"metadata":{"tags":[],"colab":{"base_uri":"https://localhost:8080/"},"id":"1X5XxzrKtWlD","executionInfo":{"status":"ok","timestamp":1693919110506,"user_tz":-60,"elapsed":9935,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"66e52d79-6b58-45f7-8675-b6d323f80927"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting Bio\n","  Downloading bio-1.5.9-py3-none-any.whl (276 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m276.4/276.4 kB\u001b[0m \u001b[31m3.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hCollecting biopython>=1.80 (from Bio)\n","  Downloading biopython-1.81-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m20.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from Bio) (2.31.0)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from Bio) (4.66.1)\n","Collecting mygene (from Bio)\n","  Downloading mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n","Requirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from Bio) (1.5.3)\n","Requirement already satisfied: pooch in /usr/local/lib/python3.10/dist-packages (from Bio) (1.7.0)\n","Collecting gprofiler-official (from Bio)\n","  Downloading gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from biopython>=1.80->Bio) (1.23.5)\n","Collecting biothings-client>=0.2.6 (from mygene->Bio)\n","  Downloading biothings_client-0.3.0-py2.py3-none-any.whl (29 kB)\n","Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.10/dist-packages (from pandas->Bio) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->Bio) (2023.3)\n","Requirement already satisfied: platformdirs>=2.5.0 in /usr/local/lib/python3.10/dist-packages (from pooch->Bio) (3.10.0)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from pooch->Bio) (23.1)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->Bio) (3.2.0)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->Bio) (3.4)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->Bio) (2.0.4)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->Bio) (2023.7.22)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.1->pandas->Bio) (1.16.0)\n","Installing collected packages: biopython, gprofiler-official, biothings-client, mygene, Bio\n","Successfully installed Bio-1.5.9 biopython-1.81 biothings-client-0.3.0 gprofiler-official-1.0.0 mygene-3.2.2\n"]}],"source":["!pip install Bio"]},{"cell_type":"code","execution_count":5,"metadata":{"tags":[],"id":"8N2CySIEtWlD","executionInfo":{"status":"ok","timestamp":1693919110506,"user_tz":-60,"elapsed":9,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["import plotly.io as pio\n","dpi = 600\n","format = \".eps\"\n","width = 1.0\n","height = 1.0\n","scale = 6\n","text_color = \"black\""]},{"cell_type":"code","execution_count":6,"metadata":{"tags":[],"colab":{"base_uri":"https://localhost:8080/"},"id":"S59dUSo7tWlD","executionInfo":{"status":"ok","timestamp":1693919110507,"user_tz":-60,"elapsed":9,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"0c9fdb55-816e-4a0e-ac6c-a1bd1670ab1f"},"outputs":[{"output_type":"stream","name":"stderr","text":["/usr/local/lib/python3.10/dist-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n","  warnings.warn(\n"]}],"source":["import tmscoring"]},{"cell_type":"code","source":["from google.colab import drive\n","drive.mount('/content/drive')"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"wKDf1CqdZn7M","executionInfo":{"status":"ok","timestamp":1693919135700,"user_tz":-60,"elapsed":25199,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"5b35eedd-1697-4fb5-dd01-c63389eeb19f"},"execution_count":7,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}]},{"cell_type":"code","source":["cd drive"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ILRzBXE3aPQh","executionInfo":{"status":"ok","timestamp":1693919135700,"user_tz":-60,"elapsed":12,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"ab0bac72-84bb-4461-bbd8-67ceb4371f63"},"execution_count":8,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive\n"]}]},{"cell_type":"code","source":["cd MyDrive"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"DLNFclbEaSB6","executionInfo":{"status":"ok","timestamp":1693919135701,"user_tz":-60,"elapsed":10,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"0fc629c3-c1d6-4f4a-b664-dfc763a6e7c4"},"execution_count":9,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive\n"]}]},{"cell_type":"code","source":["cd XBBvariants"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"HsFCN7KWaYbG","executionInfo":{"status":"ok","timestamp":1693919135701,"user_tz":-60,"elapsed":8,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"0a85dce5-34cc-4a1a-93ef-aebc31b0d512"},"execution_count":10,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/XBBvariants\n"]}]},{"cell_type":"code","source":["import os\n","cwd = os.getcwd() + os.sep\n","cwd"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":36},"id":"J84lO1fiajIR","executionInfo":{"status":"ok","timestamp":1693919135701,"user_tz":-60,"elapsed":7,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"847d9a4f-fbf1-4ee2-a5f6-ad7d27b8d8df"},"execution_count":11,"outputs":[{"output_type":"execute_result","data":{"text/plain":["'/content/drive/MyDrive/XBBvariants/'"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"}},"metadata":{},"execution_count":11}]},{"cell_type":"code","execution_count":15,"metadata":{"tags":[],"id":"43686ccd-9e38-4270-8fe9-907c5e73e0f2","executionInfo":{"status":"ok","timestamp":1693919139329,"user_tz":-60,"elapsed":487,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["spike_muts_variants = {\n","                        \"XBB.1\": np.array([19, 24, 83, 142, 146, 183, 213, 252, 339, 346, 368, 371, 373, 375, 376, 405, 408, 417, 440, 445, 446,\n","                                              460, 477, 478, 484, 490, 498, 501, 505, 614, 655, 679, 681, 764, 796, 954, 969]),\n","                        \"XBB.1.9.1\": np.array([19, 24, 83, 142, 146, 183, 213, 252, 339, 346, 368, 371, 373, 375, 376, 405, 408, 417, 440, 445, 446,\n","                                              460, 477, 478, 484, 486, 490, 498, 501, 505, 614, 655, 679, 681, 764, 796, 954, 969]),\n","                       \"XBB.2.3\": np.array([19, 24, 83, 142, 146, 183, 213, 253, 339, 346, 368, 371, 373, 375, 376, 405, 408, 417, 440, 445, 446,\n","                                              460, 477, 478, 484, 486, 490, 498, 501, 505, 521, 614, 655, 679, 681, 764, 796, 954, 969]),\n","                       \"XBB.1.5\": np.array([19, 24, 83, 142, 146, 183, 213, 252, 339, 346, 368, 371, 373, 375, 376, 405, 408, 417, 440, 445, 446,\n","                                              460, 477, 478, 484, 486, 490, 498, 501, 505, 614, 655, 679, 681, 764, 796, 954, 969]),\n","                       \"XBB.1.16\": np.array([19, 24, 83, 142, 146, 180, 183, 213, 252, 339, 346, 368, 371, 373, 375, 376, 405, 408, 417, 440, 445, 446,\n","                                              460, 477, 478, 484, 486, 490, 498, 501, 505, 614, 655, 679, 681, 764, 796, 954, 969]),\n","                       \"EG.5.1\": np.array([19, 24, 52, 83, 142, 146, 180, 183, 213, 252, 339, 346, 368, 371, 373, 375, 376, 405, 408, 417, 440, 445, 446,\n","                                            456, 460, 477, 478, 484, 486, 490, 498, 501, 505, 614, 655, 679, 681, 764, 796, 954, 969])\n","                      }"]},{"cell_type":"code","source":["conformations = [\"closed\", \"open\", \"complex\"]"],"metadata":{"id":"KXZa7db_cVW8","executionInfo":{"status":"ok","timestamp":1693919140935,"user_tz":-60,"elapsed":6,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"execution_count":16,"outputs":[]},{"cell_type":"code","execution_count":17,"metadata":{"id":"_gFduGRoIZYt","tags":[],"executionInfo":{"status":"ok","timestamp":1693919142639,"user_tz":-60,"elapsed":2,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["th_min = 4.0\n","th_max = 8.0\n","\n","basepath = os.getcwd()+\"/Data/\"\n","pdbFilesPath = os.getcwd()+\"/PDBs/\""]},{"cell_type":"code","execution_count":18,"metadata":{"id":"tDJ_Bf_2id05","tags":[],"executionInfo":{"status":"ok","timestamp":1693919143896,"user_tz":-60,"elapsed":1,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["def printProgressBar (iteration, total):\n","\n","    length = 100\n","    fill = '█'\n","    printEnd = '\\r'\n","    percent = (\"{0:.\" + str(2) + \"f}\").format(100 * (iteration / float(total)))\n","    filledLength = int(length * iteration // total)\n","    bar = fill * filledLength + '-' * (length - filledLength)\n","    print(f\"\\r |{bar}| Current progress: {percent}%\", end = printEnd)\n","    sys.stdout.flush()\n","\n","    if percent == 100:\n","        print()"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"lhzYnUtTkMba","tags":[]},"outputs":[],"source":["def adjacent_matrix(output_path, coordinates, min_=4, max_=8, mutations = None):\n","    from sklearn.metrics import pairwise_distances\n","    \"\"\"\n","    Parallel computation of the adjacency matrix.\n","    Parameters:\n","        output_path: string, is the output file path.\n","        coordinates: np.array, contains the euclidean distance between the alpha carbon of the amino acids of the proteins.\n","        min_: float, is the minimum threshold distance for extract only the non covalent interactions between amino acids.\n","        max_: float, is the maximum threshold distance for extract only the significant interactions between amino acids.\n","    Returns: adj\n","    \"\"\"\n","\n","    n = coordinates.shape[0]\n","    cords = np.hstack(coordinates[:,1]).reshape(n,3)\n","    adj = np.zeros((n,n))\n","    #d = np.zeros((n,n), dtype=float)\n","    d = pairwise_distances(X = cords, n_jobs = -1)\n","\n","    value = 0\n","    printProgressBar(value, n)\n","\n","    for i in range(n):\n","        for j in range(i):\n","            if ((d[i][j]>min_) and (d[i][j]<max_)):\n","                adj[i][j] = 1\n","                adj[j][i] = 1\n","\n","        printProgressBar(i + 1, n)\n","\n","    printProgressBar(i + 1, n)\n","\n","\n","    if not os.path.exists(\"{}Adj_Swiss\".format(output_path)):\n","        os.makedirs(\"{}Adj_Swiss\".format(output_path))\n","    if mutations is not None:\n","                # Save\n","        if isinstance(mutations, list):\n","            if len(mutations)>1:\n","                str_mut = \"\"\n","                for mut in mutations:\n","                    str_mut += \"{}_\".format(mut)\n","                str_mut = str_mut[:-2]\n","            else:\n","                str_mut = mutations[0]\n","        else:\n","            str_mut = mutations\n","        np.savetxt(\"{}Adj_Swiss\\\\6VXX_{}_adj_mat_{}_{}.txt\".format(output_path, str_mut, min_, max_), adj, fmt='%.2f')\n","    else:\n","        np.savetxt(\"{}Adj_Swiss\\\\6VXX_adj_mat_{}_{}.txt\".format(output_path, min_, max_), adj, fmt='%.2f')\n","    print(\"saved adj matrix\")\n","\n","    return adj"]},{"cell_type":"code","execution_count":19,"metadata":{"id":"fmC58tJUmocU","tags":[],"executionInfo":{"status":"ok","timestamp":1693919145740,"user_tz":-60,"elapsed":1,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["def readPDBFile(pdbFilePath):\n","    \"\"\"\n","    Read the pdb file.\n","    Parameters:\n","        pdbFilePath: string, is the complete PDB file path to read.\n","    Returns:\n","        atoms: np.array, contains all the informations contained in the 'ATOM' key of the pdb files.\n","    \"\"\"\n","\n","    atoms = []\n","    with open(pdbFilePath) as pdbfile:\n","        for line in pdbfile:\n","            if line[:4] == 'ATOM':\n","                #split the line\n","                splitted_line = [line[:6], line[6:11], line[12:16], line[17:20], line[21], line[22:26], line[30:38], line[38:46], line[46:54], line[56:61], line[62:66]]\n","                atoms.append(splitted_line)\n","    return np.array(atoms)"]},{"cell_type":"code","execution_count":20,"metadata":{"id":"sk1-zTJDkOhB","tags":[],"executionInfo":{"status":"ok","timestamp":1693919148616,"user_tz":-60,"elapsed":4,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["def getResidueCoordinates(atoms):\n","    \"\"\"\n","    Compute the distances between the alpha carbons of the amino acids.\n","    Parameters:\n","        atoms: np.array, contains all the informations contained in the 'ATOM' key of the pdb files.\n","    Returns:\n","        coordinates: np.array, contains alpha carbon coordinates of the amino acids of the proteins.\n","    \"\"\"\n","    coordinates = []\n","    residues_list = ['ALA', 'CYS', 'ASP', 'GLU', 'PHE', 'GLY', 'HIS', 'ILE', 'LYS',\n","                   'LEU', 'MET', 'ASN', 'PRO', 'GLN', 'ARG', 'SER', 'THR', 'VAL',\n","                   'TRP', 'TYR']\n","\n","    dropped = []\n","\n","    last_residue_num = 0\n","\n","    for i, atom in enumerate(atoms):\n","\n","        residue_name = atom[3]\n","        residue_chain = atom[4]\n","        residue_num = int (atom[5])\n","\n","        if residue_name in residues_list:\n","\n","            if (residue_num != last_residue_num):\n","\n","                if (atom[2].replace(\" \",\"\")==\"CA\"):\n","                    cord_C = atom[6:9]\n","                    coordinates.append([residue_name + str (residue_num) + \" \" + residue_chain, cord_C])\n","                    last_residue_num = residue_num\n","\n","        else:\n","            dropped.append(residue_name)\n","\n","    return np.array(coordinates)\n"]},{"cell_type":"code","execution_count":21,"metadata":{"tags":[],"id":"mEKVhya-tWlH","executionInfo":{"status":"ok","timestamp":1693919149771,"user_tz":-60,"elapsed":1,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["def compute_tm_score(pdbWTpath, pdbFilePath):\n","    alignment = tmscoring.TMscoring(pdbWTpath, pdbFilePath)\n","    # Find the optimal alignment\n","    alignment.optimise()\n","    # Get the TM score:\n","    tm_score = alignment.tmscore(**alignment.get_current_values())\n","    return tm_score"]},{"cell_type":"code","execution_count":null,"metadata":{"tags":[],"colab":{"base_uri":"https://localhost:8080/","height":423},"id":"LCQtFywQtWlI","executionInfo":{"status":"ok","timestamp":1693588337784,"user_tz":-120,"elapsed":1165,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"796b361f-4b3e-4a00-b303-24c852cd3e0c"},"outputs":[{"output_type":"execute_result","data":{"text/plain":["      Chain Wild type AA  Residue ID Amino acid Overall Stability  \\\n","0         A          GLN          14        GLY     Destabilising   \n","1         A          GLN          14        ALA     Destabilising   \n","2         A          GLN          14        VAL     Destabilising   \n","3         A          GLN          14        LEU     Destabilising   \n","4         A          GLN          14        ILE     Destabilising   \n","...     ...          ...         ...        ...               ...   \n","64462     C          GLU        1144        ASN     Destabilising   \n","64463     C          GLU        1144        CYS       Stabilising   \n","64464     C          GLU        1144        ASP     Destabilising   \n","64465     C          GLU        1144        ARG     Destabilising   \n","64466     C          GLU        1144        HIS     Destabilising   \n","\n","            Torsion  Predicted DDG (kcal/mol)  \n","0        Favourable                     -1.35  \n","1        Favourable                     -0.93  \n","2      Unfavourable                     -1.64  \n","3        Favourable                     -1.52  \n","4      Unfavourable                     -1.80  \n","...             ...                       ...  \n","64462    Favourable                     -1.90  \n","64463  Unfavourable                      0.10  \n","64464    Favourable                     -0.97  \n","64465  Unfavourable                     -1.01  \n","64466    Favourable                     -3.75  \n","\n","[64467 rows x 7 columns]"],"text/html":["\n","  <div id=\"df-95798141-22cc-4af7-881d-8d93866bb9d0\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>Chain</th>\n","      <th>Wild type AA</th>\n","      <th>Residue ID</th>\n","      <th>Amino acid</th>\n","      <th>Overall Stability</th>\n","      <th>Torsion</th>\n","      <th>Predicted DDG (kcal/mol)</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>A</td>\n","      <td>GLN</td>\n","      <td>14</td>\n","      <td>GLY</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.35</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>A</td>\n","      <td>GLN</td>\n","      <td>14</td>\n","      <td>ALA</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.93</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>A</td>\n","      <td>GLN</td>\n","      <td>14</td>\n","      <td>VAL</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-1.64</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>A</td>\n","      <td>GLN</td>\n","      <td>14</td>\n","      <td>LEU</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.52</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>A</td>\n","      <td>GLN</td>\n","      <td>14</td>\n","      <td>ILE</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-1.80</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>64462</th>\n","      <td>C</td>\n","      <td>GLU</td>\n","      <td>1144</td>\n","      <td>ASN</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.90</td>\n","    </tr>\n","    <tr>\n","      <th>64463</th>\n","      <td>C</td>\n","      <td>GLU</td>\n","      <td>1144</td>\n","      <td>CYS</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.10</td>\n","    </tr>\n","    <tr>\n","      <th>64464</th>\n","      <td>C</td>\n","      <td>GLU</td>\n","      <td>1144</td>\n","      <td>ASP</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.97</td>\n","    </tr>\n","    <tr>\n","      <th>64465</th>\n","      <td>C</td>\n","      <td>GLU</td>\n","      <td>1144</td>\n","      <td>ARG</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-1.01</td>\n","    </tr>\n","    <tr>\n","      <th>64466</th>\n","      <td>C</td>\n","      <td>GLU</td>\n","      <td>1144</td>\n","      <td>HIS</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-3.75</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>64467 rows × 7 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-95798141-22cc-4af7-881d-8d93866bb9d0')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-95798141-22cc-4af7-881d-8d93866bb9d0 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-95798141-22cc-4af7-881d-8d93866bb9d0');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","<div id=\"df-5aac0921-9a03-44a2-92f5-cabd25bb8204\">\n","  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-5aac0921-9a03-44a2-92f5-cabd25bb8204')\"\n","            title=\"Suggest charts.\"\n","            style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","  </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","  <script>\n","    async function quickchart(key) {\n","      const quickchartButtonEl =\n","        document.querySelector('#' + key + ' button');\n","      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","      quickchartButtonEl.classList.add('colab-df-spinner');\n","      try {\n","        const charts = await google.colab.kernel.invokeFunction(\n","            'suggestCharts', [key], {});\n","      } catch (error) {\n","        console.error('Error during call to suggestCharts:', error);\n","      }\n","      quickchartButtonEl.classList.remove('colab-df-spinner');\n","      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","    }\n","    (() => {\n","      let quickchartButtonEl =\n","        document.querySelector('#df-5aac0921-9a03-44a2-92f5-cabd25bb8204 button');\n","      quickchartButtonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","    })();\n","  </script>\n","</div>\n","    </div>\n","  </div>\n"]},"metadata":{},"execution_count":26}],"source":["import pandas as pd\n","\n","df = pd.read_csv(basepath+\"cupsat_prediction_6vxxpdb_thermal_8F38B585EA3FF7EEDD431875ADDACDF6.csv\", sep = ';', header = 0)\n","df"]},{"cell_type":"code","execution_count":null,"metadata":{"tags":[],"id":"zrgkmIRdtWlI"},"outputs":[],"source":["df_groupped = df.groupby(\"Residue ID\")"]},{"cell_type":"code","execution_count":null,"metadata":{"tags":[],"colab":{"base_uri":"https://localhost:8080/","height":1000},"id":"WLG2eubttWlI","executionInfo":{"status":"ok","timestamp":1693588345405,"user_tz":-120,"elapsed":5,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"9fd9163a-b85f-4bb0-f094-926345a0841e"},"outputs":[{"output_type":"execute_result","data":{"text/plain":["      Chain Wild type AA  Residue ID Amino acid Overall Stability  \\\n","8968      A          PHE         486        GLY     Destabilising   \n","8969      A          PHE         486        ALA         No change   \n","8970      A          PHE         486        VAL     Destabilising   \n","8971      A          PHE         486        LEU     Destabilising   \n","8972      A          PHE         486        ILE       Stabilising   \n","8973      A          PHE         486        MET     Destabilising   \n","8974      A          PHE         486        PRO       Stabilising   \n","8975      A          PHE         486        TRP       Stabilising   \n","8976      A          PHE         486        SER       Stabilising   \n","8977      A          PHE         486        THR     Destabilising   \n","8978      A          PHE         486        GLN     Destabilising   \n","8979      A          PHE         486        LYS       Stabilising   \n","8980      A          PHE         486        TYR     Destabilising   \n","8981      A          PHE         486        ASN     Destabilising   \n","8982      A          PHE         486        CYS     Destabilising   \n","8983      A          PHE         486        GLU     Destabilising   \n","8984      A          PHE         486        ASP       Stabilising   \n","8985      A          PHE         486        ARG       Stabilising   \n","8986      A          PHE         486        HIS     Destabilising   \n","30457     B          PHE         486        GLY       Stabilising   \n","30458     B          PHE         486        ALA       Stabilising   \n","30459     B          PHE         486        VAL     Destabilising   \n","30460     B          PHE         486        LEU     Destabilising   \n","30461     B          PHE         486        ILE       Stabilising   \n","30462     B          PHE         486        MET     Destabilising   \n","30463     B          PHE         486        PRO       Stabilising   \n","30464     B          PHE         486        TRP     Destabilising   \n","30465     B          PHE         486        SER       Stabilising   \n","30466     B          PHE         486        THR       Stabilising   \n","30467     B          PHE         486        GLN       Stabilising   \n","30468     B          PHE         486        LYS       Stabilising   \n","30469     B          PHE         486        TYR     Destabilising   \n","30470     B          PHE         486        ASN       Stabilising   \n","30471     B          PHE         486        CYS       Stabilising   \n","30472     B          PHE         486        GLU       Stabilising   \n","30473     B          PHE         486        ASP       Stabilising   \n","30474     B          PHE         486        ARG       Stabilising   \n","30475     B          PHE         486        HIS     Destabilising   \n","51946     C          PHE         486        GLY       Stabilising   \n","51947     C          PHE         486        ALA       Stabilising   \n","51948     C          PHE         486        VAL       Stabilising   \n","51949     C          PHE         486        LEU     Destabilising   \n","51950     C          PHE         486        ILE       Stabilising   \n","51951     C          PHE         486        MET     Destabilising   \n","51952     C          PHE         486        PRO       Stabilising   \n","51953     C          PHE         486        TRP       Stabilising   \n","51954     C          PHE         486        SER       Stabilising   \n","51955     C          PHE         486        THR       Stabilising   \n","51956     C          PHE         486        GLN       Stabilising   \n","51957     C          PHE         486        LYS       Stabilising   \n","51958     C          PHE         486        TYR       Stabilising   \n","51959     C          PHE         486        ASN       Stabilising   \n","51960     C          PHE         486        CYS       Stabilising   \n","51961     C          PHE         486        GLU       Stabilising   \n","51962     C          PHE         486        ASP       Stabilising   \n","51963     C          PHE         486        ARG       Stabilising   \n","51964     C          PHE         486        HIS       Stabilising   \n","\n","            Torsion  Predicted DDG (kcal/mol)  \n","8968   Unfavourable                     -0.99  \n","8969     Favourable                     -0.00  \n","8970     Favourable                     -0.98  \n","8971     Favourable                     -1.52  \n","8972     Favourable                      0.50  \n","8973     Favourable                     -3.30  \n","8974   Unfavourable                      0.17  \n","8975     Favourable                      0.14  \n","8976   Unfavourable                      0.29  \n","8977   Unfavourable                     -0.03  \n","8978     Favourable                     -0.07  \n","8979     Favourable                      0.29  \n","8980     Favourable                     -1.20  \n","8981   Unfavourable                     -0.63  \n","8982   Unfavourable                     -0.08  \n","8983     Favourable                     -0.05  \n","8984   Unfavourable                      0.37  \n","8985     Favourable                      0.94  \n","8986   Unfavourable                     -0.61  \n","30457  Unfavourable                      0.98  \n","30458    Favourable                      0.87  \n","30459    Favourable                     -0.17  \n","30460    Favourable                     -1.15  \n","30461    Favourable                      0.17  \n","30462    Favourable                     -1.66  \n","30463  Unfavourable                      1.03  \n","30464    Favourable                     -0.51  \n","30465  Unfavourable                      1.32  \n","30466  Unfavourable                      0.90  \n","30467    Favourable                      1.07  \n","30468    Favourable                      1.48  \n","30469  Unfavourable                     -0.20  \n","30470  Unfavourable                      0.68  \n","30471  Unfavourable                      5.65  \n","30472    Favourable                      1.16  \n","30473  Unfavourable                      1.27  \n","30474    Favourable                      1.68  \n","30475  Unfavourable                     -0.12  \n","51946  Unfavourable                      1.26  \n","51947    Favourable                      1.09  \n","51948    Favourable                      0.39  \n","51949    Favourable                     -0.92  \n","51950    Favourable                      0.56  \n","51951    Favourable                     -0.93  \n","51952  Unfavourable                      0.90  \n","51953    Favourable                      1.09  \n","51954  Unfavourable                      1.27  \n","51955  Unfavourable                      1.01  \n","51956    Favourable                      1.18  \n","51957    Favourable                      1.53  \n","51958    Favourable                      0.20  \n","51959  Unfavourable                      0.82  \n","51960  Unfavourable                      3.80  \n","51961    Favourable                      1.32  \n","51962  Unfavourable                      1.09  \n","51963    Favourable                      1.95  \n","51964  Unfavourable                      0.19  "],"text/html":["\n","  <div id=\"df-80936ea6-9e11-49e6-8231-74c63f6131f9\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>Chain</th>\n","      <th>Wild type AA</th>\n","      <th>Residue ID</th>\n","      <th>Amino acid</th>\n","      <th>Overall Stability</th>\n","      <th>Torsion</th>\n","      <th>Predicted DDG (kcal/mol)</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>8968</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLY</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.99</td>\n","    </tr>\n","    <tr>\n","      <th>8969</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ALA</td>\n","      <td>No change</td>\n","      <td>Favourable</td>\n","      <td>-0.00</td>\n","    </tr>\n","    <tr>\n","      <th>8970</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>VAL</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.98</td>\n","    </tr>\n","    <tr>\n","      <th>8971</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>LEU</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.52</td>\n","    </tr>\n","    <tr>\n","      <th>8972</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ILE</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.50</td>\n","    </tr>\n","    <tr>\n","      <th>8973</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>MET</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-3.30</td>\n","    </tr>\n","    <tr>\n","      <th>8974</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>PRO</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.17</td>\n","    </tr>\n","    <tr>\n","      <th>8975</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>TRP</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.14</td>\n","    </tr>\n","    <tr>\n","      <th>8976</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>SER</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.29</td>\n","    </tr>\n","    <tr>\n","      <th>8977</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>THR</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.03</td>\n","    </tr>\n","    <tr>\n","      <th>8978</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLN</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.07</td>\n","    </tr>\n","    <tr>\n","      <th>8979</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>LYS</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.29</td>\n","    </tr>\n","    <tr>\n","      <th>8980</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>TYR</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.20</td>\n","    </tr>\n","    <tr>\n","      <th>8981</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ASN</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.63</td>\n","    </tr>\n","    <tr>\n","      <th>8982</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>CYS</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.08</td>\n","    </tr>\n","    <tr>\n","      <th>8983</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLU</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.05</td>\n","    </tr>\n","    <tr>\n","      <th>8984</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ASP</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.37</td>\n","    </tr>\n","    <tr>\n","      <th>8985</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ARG</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.94</td>\n","    </tr>\n","    <tr>\n","      <th>8986</th>\n","      <td>A</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>HIS</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.61</td>\n","    </tr>\n","    <tr>\n","      <th>30457</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLY</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.98</td>\n","    </tr>\n","    <tr>\n","      <th>30458</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ALA</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.87</td>\n","    </tr>\n","    <tr>\n","      <th>30459</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>VAL</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.17</td>\n","    </tr>\n","    <tr>\n","      <th>30460</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>LEU</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.15</td>\n","    </tr>\n","    <tr>\n","      <th>30461</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ILE</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.17</td>\n","    </tr>\n","    <tr>\n","      <th>30462</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>MET</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-1.66</td>\n","    </tr>\n","    <tr>\n","      <th>30463</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>PRO</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.03</td>\n","    </tr>\n","    <tr>\n","      <th>30464</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>TRP</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.51</td>\n","    </tr>\n","    <tr>\n","      <th>30465</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>SER</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.32</td>\n","    </tr>\n","    <tr>\n","      <th>30466</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>THR</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.90</td>\n","    </tr>\n","    <tr>\n","      <th>30467</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLN</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.07</td>\n","    </tr>\n","    <tr>\n","      <th>30468</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>LYS</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.48</td>\n","    </tr>\n","    <tr>\n","      <th>30469</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>TYR</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.20</td>\n","    </tr>\n","    <tr>\n","      <th>30470</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ASN</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.68</td>\n","    </tr>\n","    <tr>\n","      <th>30471</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>CYS</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>5.65</td>\n","    </tr>\n","    <tr>\n","      <th>30472</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLU</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.16</td>\n","    </tr>\n","    <tr>\n","      <th>30473</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ASP</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.27</td>\n","    </tr>\n","    <tr>\n","      <th>30474</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ARG</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.68</td>\n","    </tr>\n","    <tr>\n","      <th>30475</th>\n","      <td>B</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>HIS</td>\n","      <td>Destabilising</td>\n","      <td>Unfavourable</td>\n","      <td>-0.12</td>\n","    </tr>\n","    <tr>\n","      <th>51946</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLY</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.26</td>\n","    </tr>\n","    <tr>\n","      <th>51947</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ALA</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.09</td>\n","    </tr>\n","    <tr>\n","      <th>51948</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>VAL</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.39</td>\n","    </tr>\n","    <tr>\n","      <th>51949</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>LEU</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.92</td>\n","    </tr>\n","    <tr>\n","      <th>51950</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ILE</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.56</td>\n","    </tr>\n","    <tr>\n","      <th>51951</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>MET</td>\n","      <td>Destabilising</td>\n","      <td>Favourable</td>\n","      <td>-0.93</td>\n","    </tr>\n","    <tr>\n","      <th>51952</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>PRO</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.90</td>\n","    </tr>\n","    <tr>\n","      <th>51953</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>TRP</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.09</td>\n","    </tr>\n","    <tr>\n","      <th>51954</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>SER</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.27</td>\n","    </tr>\n","    <tr>\n","      <th>51955</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>THR</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.01</td>\n","    </tr>\n","    <tr>\n","      <th>51956</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLN</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.18</td>\n","    </tr>\n","    <tr>\n","      <th>51957</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>LYS</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.53</td>\n","    </tr>\n","    <tr>\n","      <th>51958</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>TYR</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>0.20</td>\n","    </tr>\n","    <tr>\n","      <th>51959</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ASN</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.82</td>\n","    </tr>\n","    <tr>\n","      <th>51960</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>CYS</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>3.80</td>\n","    </tr>\n","    <tr>\n","      <th>51961</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>GLU</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.32</td>\n","    </tr>\n","    <tr>\n","      <th>51962</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ASP</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>1.09</td>\n","    </tr>\n","    <tr>\n","      <th>51963</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>ARG</td>\n","      <td>Stabilising</td>\n","      <td>Favourable</td>\n","      <td>1.95</td>\n","    </tr>\n","    <tr>\n","      <th>51964</th>\n","      <td>C</td>\n","      <td>PHE</td>\n","      <td>486</td>\n","      <td>HIS</td>\n","      <td>Stabilising</td>\n","      <td>Unfavourable</td>\n","      <td>0.19</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-80936ea6-9e11-49e6-8231-74c63f6131f9')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-80936ea6-9e11-49e6-8231-74c63f6131f9 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-80936ea6-9e11-49e6-8231-74c63f6131f9');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","<div id=\"df-0307091d-40c7-49bd-a4a0-6d119647a0bd\">\n","  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-0307091d-40c7-49bd-a4a0-6d119647a0bd')\"\n","            title=\"Suggest charts.\"\n","            style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","  </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","  <script>\n","    async function quickchart(key) {\n","      const quickchartButtonEl =\n","        document.querySelector('#' + key + ' button');\n","      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","      quickchartButtonEl.classList.add('colab-df-spinner');\n","      try {\n","        const charts = await google.colab.kernel.invokeFunction(\n","            'suggestCharts', [key], {});\n","      } catch (error) {\n","        console.error('Error during call to suggestCharts:', error);\n","      }\n","      quickchartButtonEl.classList.remove('colab-df-spinner');\n","      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","    }\n","    (() => {\n","      let quickchartButtonEl =\n","        document.querySelector('#df-0307091d-40c7-49bd-a4a0-6d119647a0bd button');\n","      quickchartButtonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","    })();\n","  </script>\n","</div>\n","    </div>\n","  </div>\n"]},"metadata":{},"execution_count":28}],"source":["resi_aa = 486\n","df_resi = df_groupped.get_group(resi_aa)\n","df_resi"]},{"cell_type":"code","execution_count":22,"metadata":{"tags":[],"id":"J7iApY-9tWlI","executionInfo":{"status":"ok","timestamp":1693919155200,"user_tz":-60,"elapsed":443,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["def get_deltaG_mutation(df_groupped, mutations):\n","\n","\n","    all_res_names = {\n","        \"A\":\"ALA\",\n","        \"C\":\"CYS\",\n","        \"D\":\"ASP\",\n","        \"E\":\"GLU\",\n","        \"F\":\"PHE\",\n","        \"G\":\"GLY\",\n","        \"H\":\"HIS\",\n","        \"I\":\"ILE\",\n","        \"K\":\"LYS\",\n","        \"L\":\"LEU\",\n","        \"M\":\"MET\",\n","        \"N\":\"ASN\",\n","        \"P\":\"PRO\",\n","        \"Q\":\"GLN\",\n","        \"R\":\"ARG\",\n","        \"S\":\"SER\",\n","        \"T\":\"THR\",\n","        \"V\":\"VAL\",\n","        \"W\":\"TRP\",\n","        \"Y\":\"TYR\"\n","    }\n","\n","    deltag = pd.DataFrame(columns = df_groupped.first().columns)\n","    if isinstance(mutations, list):\n","        for mutation in mutations:\n","            to_aa = mutation[-1]\n","            from_aa = mutation[0]\n","            resi_aa = int(mutation[1:-1])\n","            df_resi = df_groupped.get_group(resi_aa)\n","            deltag = pd.concat([deltag, df_resi[df_resi[\"Amino acid\"]==all_res_names[to_aa]]])\n","    return deltag"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":850},"id":"65vT_viCOP3H","outputId":"a87cff8d-392f-4602-9bf7-f20d64f1461c"},"outputs":[{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Reading WT adj matrix\n","Reading pdb file with mutations:  A67V\n","Reading WT adj matrix with mutations:  ['A67V']\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation A67V vs WT. TM score: 0.9999999983740927\n","WT + mutation A67V vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","1008      A          ALA        VAL       Stabilising  Favourable   \n","22497     B          ALA        VAL       Stabilising  Favourable   \n","43986     C          ALA        VAL       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","1008                      2.67        67.0  \n","22497                     2.68        67.0  \n","43986                     3.12        67.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+A67V Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  T95I\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Reading WT adj matrix with mutations:  ['T95I']\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation T95I vs WT. TM score: 0.9999999907144737\n","WT + mutation T95I vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","1543      A          THR        ILE       Stabilising  Favourable   \n","23032     B          THR        ILE       Stabilising  Favourable   \n","44521     C          THR        ILE       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","1543                       0.6        95.0  \n","23032                     0.54        95.0  \n","44521                     0.14        95.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+T95I Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  G142D\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Reading WT adj matrix with mutations:  ['G142D']\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation G142D vs WT. TM score: 0.9999999960069327\n","WT + mutation G142D vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","2448      A          GLY        ASP     Destabilising  Unfavourable   \n","23937     B          GLY        ASP     Destabilising  Unfavourable   \n","45426     C          GLY        ASP     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","2448                      -0.3       142.0  \n","23937                     -0.2       142.0  \n","45426                    -0.45       142.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+G142D Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  L212I\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation L212I vs WT. TM score: 0.9999999977710078\n","WT + mutation L212I vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","3765      A          LEU        ILE     Destabilising  Favourable   \n","25254     B          LEU        ILE     Destabilising  Favourable   \n","46743     C          LEU        ILE     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","3765                     -0.77       212.0  \n","25254                    -0.77       212.0  \n","46743                    -0.73       212.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+L212I Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  G339D\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation G339D vs WT. TM score: 0.9999999961524518\n","WT + mutation G339D vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","6191      A          GLY        ASP       Stabilising  Unfavourable   \n","27680     B          GLY        ASP       Stabilising  Unfavourable   \n","49169     C          GLY        ASP       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","6191                      0.81       339.0  \n","27680                     0.62       339.0  \n","49169                     0.82       339.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+G339D Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  S371L\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation S371L vs WT. TM score: 0.9999999890073352\n","WT + mutation S371L vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","6786      A          SER        LEU       Stabilising  Favourable   \n","28275     B          SER        LEU       Stabilising  Favourable   \n","49764     C          SER        LEU       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","6786                      1.32       371.0  \n","28275                     1.38       371.0  \n","49764                     1.36       371.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+S371L Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  S373P\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation S373P vs WT. TM score: 0.9999999723868012\n","WT + mutation S373P vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","6827      A          SER        PRO     Destabilising  Unfavourable   \n","28316     B          SER        PRO     Destabilising  Unfavourable   \n","49805     C          SER        PRO     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","6827                     -2.36       373.0  \n","28316                    -2.44       373.0  \n","49805                    -2.46       373.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+S373P Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  S375F\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation S375F vs WT. TM score: 0.9999999979073556\n","WT + mutation S375F vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","6868      A          SER        PHE     Destabilising  Favourable   \n","28357     B          SER        PHE     Destabilising  Favourable   \n","49846     C          SER        PHE     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","6868                     -0.42       375.0  \n","28357                    -0.43       375.0  \n","49846                    -1.41       375.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+S375F Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  K417N\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation K417N vs WT. TM score: 0.9999999907302894\n","WT + mutation K417N vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","7670      A          LYS        ASN     Destabilising  Unfavourable   \n","29159     B          LYS        ASN     Destabilising  Unfavourable   \n","50648     C          LYS        ASN     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","7670                     -4.11       417.0  \n","29159                    -4.16       417.0  \n","50648                    -4.23       417.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+K417N Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  N440K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation N440K vs WT. TM score: 0.9999999904468606\n","WT + mutation N440K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","8106      A          ASN        LYS       Stabilising  Favourable   \n","29595     B          ASN        LYS       Stabilising  Favourable   \n","51084     C          ASN        LYS       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8106                      0.42       440.0  \n","29595                     0.42       440.0  \n","51084                     0.42       440.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+N440K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  G446S\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation G446S vs WT. TM score: 0.999999997618117\n","WT + mutation G446S vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","8215      A          GLY        SER     Destabilising  Favourable   \n","29704     B          GLY        SER     Destabilising  Favourable   \n","51193     C          GLY        SER     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8215                     -0.02       446.0  \n","29704                    -0.02       446.0  \n","51193                    -0.02       446.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+G446S Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  S477N\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation S477N vs WT. TM score: 0.9999999966937686\n","WT + mutation S477N vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","8810      A          SER        ASN       Stabilising  Unfavourable   \n","30299     B          SER        ASN       Stabilising  Unfavourable   \n","51788     C          SER        ASN       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8810                      0.78       477.0  \n","30299                     0.79       477.0  \n","51788                     0.94       477.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+S477N Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  T478K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation T478K vs WT. TM score: 0.9999999931416701\n","WT + mutation T478K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","8827      A          THR        LYS     Destabilising    Favourable   \n","30316     B          THR        LYS     Destabilising    Favourable   \n","51805     C          THR        LYS     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8827                     -0.51       478.0  \n","30316                    -0.87       478.0  \n","51805                    -0.83       478.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+T478K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  E484A\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation E484A vs WT. TM score: 0.9999999923630625\n","WT + mutation E484A vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","8931      A          GLU        ALA     Destabilising  Unfavourable   \n","30420     B          GLU        ALA     Destabilising    Favourable   \n","51909     C          GLU        ALA     Destabilising    Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8931                      -0.4       484.0  \n","30420                    -0.25       484.0  \n","51909                    -0.29       484.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+E484A Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  Q493R\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation Q493R vs WT. TM score: 0.999999997017091\n","WT + mutation Q493R vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","9118      A          GLN        ARG     Destabilising  Unfavourable   \n","30607     B          GLN        ARG       Stabilising  Unfavourable   \n","52096     C          GLN        ARG       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","9118                     -0.26       493.0  \n","30607                     0.03       493.0  \n","52096                      0.1       493.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+Q493R Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  G496S\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation G496S vs WT. TM score: 0.9999999997040274\n","WT + mutation G496S vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","9165      A          GLY        SER       Stabilising  Unfavourable   \n","30654     B          GLY        SER       Stabilising  Unfavourable   \n","52143     C          GLY        SER       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","9165                      0.08       496.0  \n","30654                     0.06       496.0  \n","52143                     0.09       496.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+G496S Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  Q498R\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation Q498R vs WT. TM score: 0.9999999834695353\n","WT + mutation Q498R vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","9213      A          GLN        ARG     Destabilising  Favourable   \n","30702     B          GLN        ARG     Destabilising  Favourable   \n","52191     C          GLN        ARG     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","9213                     -0.22       498.0  \n","30702                    -0.48       498.0  \n","52191                    -0.29       498.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+Q498R Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  N501Y\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation N501Y vs WT. TM score: 0.9999999924012333\n","WT + mutation N501Y vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","9266      A          ASN        TYR     Destabilising  Unfavourable   \n","30755     B          ASN        TYR     Destabilising  Unfavourable   \n","52244     C          ASN        TYR     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","9266                     -0.95       501.0  \n","30755                    -0.95       501.0  \n","52244                    -0.94       501.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+N501Y Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  Y505H\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation Y505H vs WT. TM score: 0.9999999908146618\n","WT + mutation Y505H vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","9347      A          TYR        HIS     Destabilising  Favourable   \n","30836     B          TYR        HIS     Destabilising  Favourable   \n","52325     C          TYR        HIS     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","9347                     -0.98       505.0  \n","30836                    -0.98       505.0  \n","52325                    -0.92       505.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+Y505H Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  T547K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation T547K vs WT. TM score: 0.9999999818997666\n","WT + mutation T547K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability    Torsion  \\\n","10138     A          THR        LYS     Destabilising  No change   \n","31627     B          THR        LYS       Stabilising  No change   \n","53116     C          THR        LYS       Stabilising  No change   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","10138                    -2.33       547.0  \n","31627                     2.75       547.0  \n","53116                     2.76       547.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+T547K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  D614G\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D614G vs WT. TM score: 0.9999999989518534\n","WT + mutation D614G vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","11400     A          ASP        GLY     Destabilising  Unfavourable   \n","32889     B          ASP        GLY     Destabilising  Unfavourable   \n","54378     C          ASP        GLY     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","11400                    -0.24       614.0  \n","32889                    -0.02       614.0  \n","54378                    -0.11       614.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D614G Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  H655Y\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation H655Y vs WT. TM score: 0.9999999921681376\n","WT + mutation H655Y vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","12192     A          HIS        TYR     Destabilising  Unfavourable   \n","33681     B          HIS        TYR     Destabilising  Unfavourable   \n","55170     C          HIS        TYR     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","12192                    -5.39       655.0  \n","33681                    -5.41       655.0  \n","55170                     -5.5       655.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+H655Y Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  N679K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation N679K vs WT. TM score: 0.9999999744044437\n","WT + mutation N679K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","12647     A          ASN        LYS     Destabilising    Favourable   \n","34136     B          ASN        LYS     Destabilising  Unfavourable   \n","55625     C          ASN        LYS     Destabilising    Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","12647                    -1.06       679.0  \n","34136                    -1.14       679.0  \n","55625                    -0.73       679.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+N679K Spike SARS-CoV-2 variants: 99.99993051000703 %\n","\n","\n","Reading pdb file with mutations:  P681H\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation P681H vs WT. TM score: 0.9999999968570992\n","WT + mutation P681H vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","12691     A          PRO        HIS       Stabilising  Favourable   \n","34180     B          PRO        HIS       Stabilising  Favourable   \n","55669     C          PRO        HIS       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","12691                     1.22       681.0  \n","34180                     2.08       681.0  \n","55669                     1.99       681.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+P681H Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  N764K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation N764K vs WT. TM score: 0.999999993113161\n","WT + mutation N764K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","14262     A          ASN        LYS       Stabilising  Favourable   \n","35751     B          ASN        LYS       Stabilising  Favourable   \n","57240     C          ASN        LYS       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","14262                     0.71       764.0  \n","35751                     0.65       764.0  \n","57240                     0.65       764.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+N764K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  D796Y\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D796Y vs WT. TM score: 0.9999999801453989\n","WT + mutation D796Y vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","14871     A          ASP        TYR     Destabilising  Unfavourable   \n","36360     B          ASP        TYR     Destabilising  Unfavourable   \n","57849     C          ASP        TYR     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","14871                    -1.18       796.0  \n","36360                    -1.18       796.0  \n","57849                    -1.19       796.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D796Y Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  N856K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation N856K vs WT. TM score: 0.9999999983762934\n","WT + mutation N856K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","16010     A          ASN        LYS     Destabilising  Unfavourable   \n","37499     B          ASN        LYS     Destabilising  Unfavourable   \n","58988     C          ASN        LYS     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","16010                    -2.08       856.0  \n","37499                    -2.06       856.0  \n","58988                    -2.07       856.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+N856K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  Q954H\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation Q954H vs WT. TM score: 0.9999999852362875\n","WT + mutation Q954H vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","17878     A          GLN        HIS       Stabilising  Unfavourable   \n","39367     B          GLN        HIS       Stabilising  Unfavourable   \n","60856     C          GLN        HIS       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","17878                     1.15       954.0  \n","39367                     1.17       954.0  \n","60856                     1.17       954.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+Q954H Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  N969K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation N969K vs WT. TM score: 0.999999992656289\n","WT + mutation N969K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","18157     A          ASN        LYS     Destabilising  Unfavourable   \n","39646     B          ASN        LYS     Destabilising  Unfavourable   \n","61135     C          ASN        LYS     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","18157                     -0.7       969.0  \n","39646                    -0.61       969.0  \n","61135                    -0.67       969.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+N969K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  L981F\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation L981F vs WT. TM score: 0.9999999940683728\n","WT + mutation L981F vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","18382     A          LEU        PHE       Stabilising  Unfavourable   \n","39871     B          LEU        PHE       Stabilising  Unfavourable   \n","61360     C          LEU        PHE       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","18382                     0.15       981.0  \n","39871                     0.17       981.0  \n","61360                     0.15       981.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+L981F Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Done mutation  ['L981F']\n","Reading pdb file with mutations:  V213G\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation V213G vs WT. TM score: 0.999999993911832\n","WT + mutation V213G vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","3781      A          VAL        GLY       Stabilising  Unfavourable   \n","25270     B          VAL        GLY       Stabilising  Unfavourable   \n","46759     C          VAL        GLY       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","3781                      1.55       213.0  \n","25270                     1.55       213.0  \n","46759                     1.56       213.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+V213G Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  S371F\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation S371F vs WT. TM score: 0.999999980748168\n","WT + mutation S371F vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","6792      A          SER        PHE       Stabilising  Unfavourable   \n","28281     B          SER        PHE       Stabilising  Unfavourable   \n","49770     C          SER        PHE       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","6792                      0.67       371.0  \n","28281                     0.94       371.0  \n","49770                      0.9       371.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+S371F Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  T376A\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation T376A vs WT. TM score: 0.9999999878635443\n","WT + mutation T376A vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","6879      A          THR        ALA     Destabilising  Unfavourable   \n","28368     B          THR        ALA     Destabilising  Unfavourable   \n","49857     C          THR        ALA     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","6879                     -1.29       376.0  \n","28368                    -1.31       376.0  \n","49857                    -1.33       376.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+T376A Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  D405N\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D405N vs WT. TM score: 0.9999999938579455\n","WT + mutation D405N vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","7443      A          ASP        ASN     Destabilising  Favourable   \n","28932     B          ASP        ASN     Destabilising  Favourable   \n","50421     C          ASP        ASN     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","7443                     -2.05       405.0  \n","28932                    -2.11       405.0  \n","50421                    -2.08       405.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D405N Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  R408S\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation R408S vs WT. TM score: 0.9999999920534935\n","WT + mutation R408S vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","7494      A          ARG        SER     Destabilising  Favourable   \n","28983     B          ARG        SER     Destabilising  Favourable   \n","50472     C          ARG        SER     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","7494                     -0.68       408.0  \n","28983                    -0.69       408.0  \n","50472                    -0.67       408.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+R408S Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  L452R\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation L452R vs WT. TM score: 0.9999999950462494\n","WT + mutation L452R vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","8339      A          LEU        ARG     Destabilising  Favourable   \n","29828     B          LEU        ARG     Destabilising  Favourable   \n","51317     C          LEU        ARG     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8339                     -0.93       452.0  \n","29828                    -0.97       452.0  \n","51317                    -0.89       452.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+L452R Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  F486V\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation F486V vs WT. TM score: 0.9999999978222781\n","WT + mutation F486V vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","8970      A          PHE        VAL     Destabilising  Favourable   \n","30459     B          PHE        VAL     Destabilising  Favourable   \n","51948     C          PHE        VAL       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8970                     -0.98       486.0  \n","30459                    -0.17       486.0  \n","51948                     0.39       486.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+F486V Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Done mutation  N969K\n","Reading pdb file with mutations:  R158G\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation R158G vs WT. TM score: 0.9999999995948623\n","WT + mutation R158G vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","2736      A          ARG        GLY     Destabilising  Unfavourable   \n","24225     B          ARG        GLY     Destabilising  Unfavourable   \n","45714     C          ARG        GLY     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","2736                     -2.27       158.0  \n","24225                    -2.38       158.0  \n","45714                    -3.07       158.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+R158G Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  P681R\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation P681R vs WT. TM score: 0.9999999957424046\n","WT + mutation P681R vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","12690     A          PRO        ARG       Stabilising  Unfavourable   \n","34179     B          PRO        ARG       Stabilising  Unfavourable   \n","55668     C          PRO        ARG       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","12690                     3.18       681.0  \n","34179                     3.91       681.0  \n","55668                     3.91       681.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+P681R Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  D950N\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D950N vs WT. TM score: 0.9999999973675998\n","WT + mutation D950N vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","17798     A          ASP        ASN     Destabilising  Unfavourable   \n","39287     B          ASP        ASN     Destabilising  Unfavourable   \n","60776     C          ASP        ASN     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","17798                    -0.75       950.0  \n","39287                     -0.8       950.0  \n","60776                    -0.75       950.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D950N Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Done mutation  ['D950N']\n","Reading pdb file with mutations:  A570D\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation A570D vs WT. TM score: 0.9999999951355926\n","WT + mutation A570D vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","10580     A          ALA        ASP     Destabilising  Favourable   \n","32069     B          ALA        ASP     Destabilising  Favourable   \n","53558     C          ALA        ASP     Destabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","10580                    -2.44       570.0  \n","32069                    -2.35       570.0  \n","53558                    -2.32       570.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+A570D Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  T716I\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation T716I vs WT. TM score: 0.9999999946579106\n","WT + mutation T716I vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","13342     A          THR        ILE       Stabilising  Unfavourable   \n","34831     B          THR        ILE     Destabilising  Unfavourable   \n","56320     C          THR        ILE       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","13342                     0.08       716.0  \n","34831                    -0.08       716.0  \n","56320                     0.14       716.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+T716I Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  S982A\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation S982A vs WT. TM score: 0.9999999806768884\n","WT + mutation S982A vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","18393     A          SER        ALA       Stabilising  Favourable   \n","39882     B          SER        ALA       Stabilising  Favourable   \n","61371     C          SER        ALA       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","18393                     4.07       982.0  \n","39882                     2.56       982.0  \n","61371                     2.61       982.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+S982A Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  D1118H\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D1118H vs WT. TM score: 0.9999999767190051\n","WT + mutation D1118H vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","20994     A          ASP        HIS     Destabilising  Unfavourable   \n","42483     B          ASP        HIS     Destabilising  Unfavourable   \n","63972     C          ASP        HIS     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","20994                    -4.33      1118.0  \n","42483                    -4.56      1118.0  \n","63972                    -4.34      1118.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D1118H Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Done mutation  ['D1118H']\n","Reading pdb file with mutations:  D80A\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D80A vs WT. TM score: 0.9999999689260414\n","WT + mutation D80A vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","1255      A          ASP        ALA     Destabilising  Unfavourable   \n","22744     B          ASP        ALA     Destabilising  Unfavourable   \n","44233     C          ASP        ALA     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","1255                     -5.67        80.0  \n","22744                    -5.31        80.0  \n","44233                    -4.94        80.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D80A Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  D215G\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D215G vs WT. TM score: 0.9999999988677463\n","WT + mutation D215G vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","3819      A          ASP        GLY       Stabilising  Unfavourable   \n","25308     B          ASP        GLY       Stabilising  Unfavourable   \n","46797     C          ASP        GLY       Stabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","3819                       0.6       215.0  \n","25308                     0.56       215.0  \n","46797                     0.52       215.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D215G Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  R246I\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation R246I vs WT. TM score: 0.9999999837373216\n","WT + mutation R246I vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","4412      A          ARG        ILE       Stabilising  Favourable   \n","25901     B          ARG        ILE       Stabilising  Favourable   \n","47390     C          ARG        ILE       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","4412                      2.12       246.0  \n","25901                     2.37       246.0  \n","47390                     1.91       246.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+R246I Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  E484K\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation E484K vs WT. TM score: 0.9999999941247394\n","WT + mutation E484K vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","8942      A          GLU        LYS     Destabilising    Favourable   \n","30431     B          GLU        LYS     Destabilising  Unfavourable   \n","51920     C          GLU        LYS     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","8942                     -0.62       484.0  \n","30431                    -0.16       484.0  \n","51920                    -0.25       484.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+E484K Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  A701V\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation A701V vs WT. TM score: 0.9999999983739001\n","WT + mutation A701V vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","13054     A          ALA        VAL       Stabilising  Favourable   \n","34543     B          ALA        VAL       Stabilising  Favourable   \n","56032     C          ALA        VAL       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","13054                     0.25       701.0  \n","34543                     0.28       701.0  \n","56032                     0.28       701.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+A701V Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Done mutation  ['A701V']\n","Reading pdb file with mutations:  D138Y\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation D138Y vs WT. TM score: 0.9999999930556666\n","WT + mutation D138Y vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","2369      A          ASP        TYR       Stabilising  Favourable   \n","23858     B          ASP        TYR       Stabilising  Favourable   \n","45347     C          ASP        TYR       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","2369                      2.05       138.0  \n","23858                     1.77       138.0  \n","45347                     2.42       138.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+D138Y Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  R190S\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation R190S vs WT. TM score: 0.9999999995606702\n","WT + mutation R190S vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","3352      A          ARG        SER     Destabilising  Unfavourable   \n","24841     B          ARG        SER     Destabilising  Unfavourable   \n","46330     C          ARG        SER     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","3352                     -4.74       190.0  \n","24841                    -4.74       190.0  \n","46330                    -4.71       190.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+R190S Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  K417T\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation K417T vs WT. TM score: 0.999999994183594\n","WT + mutation K417T vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability       Torsion  \\\n","7666      A          LYS        THR     Destabilising  Unfavourable   \n","29155     B          LYS        THR     Destabilising  Unfavourable   \n","50644     C          LYS        THR     Destabilising  Unfavourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","7666                     -1.86       417.0  \n","29155                    -1.89       417.0  \n","50644                    -1.89       417.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+K417T Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Reading pdb file with mutations:  T1027I\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation T1027I vs WT. TM score: 0.9999999966068084\n","WT + mutation T1027I vs WT. DeltaG stability score:       Chain Wild type AA Amino acid Overall Stability     Torsion  \\\n","19251     A          THR        ILE       Stabilising  Favourable   \n","40740     B          THR        ILE       Stabilising  Favourable   \n","62229     C          THR        ILE       Stabilising  Favourable   \n","\n","      Predicted DDG (kcal/mol)  Residue ID  \n","19251                      1.0      1027.0  \n","40740                     0.98      1027.0  \n","62229                     1.34      1027.0   vs Empty DataFrame\n","Columns: [Chain, Wild type AA, Amino acid, Overall Stability, Torsion, Predicted DDG (kcal/mol)]\n","Index: []\n","Contact similarity between WT and WT+T1027I Spike SARS-CoV-2 variants: 100.0 %\n","\n","\n","Done mutation  ['T1027I']\n","Reading pdb file with mutations:  V1176F\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\1545725321.py:36: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n","  return np.array(coordinates)\n"]},{"name":"stdout","output_type":"stream","text":["Computing WT adj matrix\n","saved adj matrix██████████████████████████████████████████████████████████████████████████████████████| Current progress: 100.00%\n"]},{"name":"stderr","output_type":"stream","text":["C:\\Users\\ugo-c\\AppData\\Local\\Temp\\ipykernel_28224\\3062668350.py:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"name":"stdout","output_type":"stream","text":["WT + mutation V1176F vs WT. TM score: 1.0\n"]},{"ename":"KeyError","evalue":"1176","output_type":"error","traceback":["\u001b[1;31m---------------------------------------------------------------------------\u001b[0m","\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)","Input \u001b[1;32mIn [63]\u001b[0m, in \u001b[0;36m<cell line: 27>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     55\u001b[0m no_mut \u001b[38;5;241m=\u001b[39m mut[\u001b[38;5;241m0\u001b[39m][\u001b[38;5;241m1\u001b[39m:\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\u001b[38;5;241m+\u001b[39mmut[\u001b[38;5;241m0\u001b[39m][\u001b[38;5;241m0\u001b[39m] \n\u001b[0;32m     57\u001b[0m deltaG_WT \u001b[38;5;241m=\u001b[39m get_deltaG_mutation(df_groupped, no_mut)\n\u001b[1;32m---> 58\u001b[0m deltaG \u001b[38;5;241m=\u001b[39m \u001b[43mget_deltaG_mutation\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdf_groupped\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmut\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     59\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWT + mutation \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m vs WT. DeltaG stability score: \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m vs \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mformat(mut[\u001b[38;5;241m0\u001b[39m], deltaG, deltaG_WT))\n\u001b[0;32m     60\u001b[0m deltaGs[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWT + mutation \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mformat(mut)] \u001b[38;5;241m=\u001b[39m deltaG\n","Input \u001b[1;32mIn [36]\u001b[0m, in \u001b[0;36mget_deltaG_mutation\u001b[1;34m(df_groupped, mutations)\u001b[0m\n\u001b[0;32m     31\u001b[0m         from_aa \u001b[38;5;241m=\u001b[39m mutation[\u001b[38;5;241m0\u001b[39m]\n\u001b[0;32m     32\u001b[0m         resi_aa \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(mutation[\u001b[38;5;241m1\u001b[39m:\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m])\n\u001b[1;32m---> 33\u001b[0m         df_resi \u001b[38;5;241m=\u001b[39m \u001b[43mdf_groupped\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_group\u001b[49m\u001b[43m(\u001b[49m\u001b[43mresi_aa\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     34\u001b[0m         deltag \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([deltag, df_resi[df_resi[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAmino acid\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m==\u001b[39mall_res_names[to_aa]]])\n\u001b[0;32m     35\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m deltag\n","File \u001b[1;32m~\\anaconda3\\envs\\PCN\\lib\\site-packages\\pandas\\core\\groupby\\groupby.py:747\u001b[0m, in \u001b[0;36mBaseGroupBy.get_group\u001b[1;34m(self, name, obj)\u001b[0m\n\u001b[0;32m    745\u001b[0m inds \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_index(name)\n\u001b[0;32m    746\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(inds):\n\u001b[1;32m--> 747\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(name)\n\u001b[0;32m    749\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m obj\u001b[38;5;241m.\u001b[39m_take_with_is_copy(inds, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maxis)\n","\u001b[1;31mKeyError\u001b[0m: 1176"]}],"source":["tm_scores = {}\n","deltaGs = {}\n","pdbWTpath =  pdbSwissFilesPath+\"6VXX.pdb\"\n","contacts_sim_mat = {}\n","figures = {}\n","contacts_sim = []\n","A_s = {}\n","atoms = readPDBFile(pdbWTpath)\n","coordinates = getResidueCoordinates(atoms)\n","if \"6VXX_adj_mat_{}_{}.txt\".format(th_min, th_max) in os.listdir(basepath+\"Adj_Swiss\"):\n","    print(\"Reading WT adj matrix\")\n","    adj_wt = read_adj_mat(basepath, \"6VXX\", th_min, th_max)\n","else:\n","    print(\"Computing WT adj matrix\")\n","    adj_wt = adjacent_matrix(basepath, coordinates, th_min, th_max, None)\n","A_s[\"WT\"] = adj_wt\n","\n","#read all done mutations from Adj Swiss directory\n","done_muts = []\n","\"\"\"\n","for filename in os.listdir(basepath+\"Adj_Swiss\"):\n","    temp = filename.split(\"_\")\n","    if temp[0] == \"6VXX\":\n","        if temp[1] != \"adj\":\n","            done_muts.append(temp[1])\n","\"\"\"\n","for spike_var, all_mutations in spike_muts_variants.items():\n","    snp_mutations = all_mutations['mut']\n","    deletions = all_mutations['del']\n","    insertions = all_mutations['ins']\n","    for mut in snp_mutations:\n","        if mut not in done_muts:\n","            mut = [mut]\n","            all_resi_spike = np.arange(len(basesequence))\n","            pdbFilePath = pdbSwissFilesPath+\"{}_{}.pdb\".format(\"6VXX\", mut[0])\n","            if \"{}_{}.pdb\".format(\"6VXX\", mut[0]) not in os.listdir(pdbSwissFilesPath):\n","                print(\"Creating pdb file with mutations: \", mut[0])\n","                create_pdb_with_mutations(pdbWTpath, mut)\n","            else:\n","                print(\"Reading pdb file with mutations: \", mut[0])\n","\n","            atoms = readPDBFile(pdbFilePath)\n","            coordinates = getResidueCoordinates(atoms)\n","            if \"6VXX_{}_adj_mat_{}_{}.txt\".format(mut[0], th_min, th_max) in os.listdir(basepath+\"Adj_Swiss\"):\n","                print(\"Reading WT adj matrix with mutations: \", mut)\n","                adj = read_adj_mat(basepath, \"6VXX\", th_min, th_max, mut[0])\n","            else:\n","                print(\"Computing WT adj matrix\")\n","                adj = adjacent_matrix(basepath, coordinates, th_min, th_max, mut[0])\n","\n","            A_s[\"WT_{}\".format(mut[0])] = adj\n","            tm_score = compute_tm_score(pdbWTpath, pdbFilePath)\n","            print(\"WT + mutation {} vs WT. TM score: {}\".format(mut[0], tm_score))\n","            tm_scores[\"WT + mutation {}\".format(mut[0])] = tm_score\n","            no_mut = mut[0][1:-1]+mut[0][0]\n","\n","            deltaG_WT = get_deltaG_mutation(df_groupped, no_mut)\n","            deltaG = get_deltaG_mutation(df_groupped, mut)\n","            print(\"WT + mutation {} vs WT. DeltaG stability score: {} vs {}\".format(mut[0], deltaG, deltaG_WT))\n","            deltaGs[\"WT + mutation {}\".format(mut)] = deltaG\n","            contact_sim_matrix = compute_contact_similarity_matrix(adj_wt, adj)\n","            contacts_sim_mat[\"WT + mutation {} vs WT\".format(mut[0])] = contact_sim_matrix\n","            contact_sim = compute_contact_similarity(contact_sim_matrix)\n","            contacts_sim.append(contact_sim)\n","            print(\"Contact similarity between WT and WT+{} Spike SARS-CoV-2 variants: {} %\".format(mut[0], contact_sim))\n","            done_muts.append(mut[0])\n","            print()\n","            print()\n","    else:\n","\n","        print(\"Done mutation \", mut)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"IpgqdUc1tWlJ","outputId":"b3e71594-def1-41d1-da19-831e47783caa"},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>tm_score</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>WT + mutation A67V</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation T95I</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation G142D</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation L212I</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation G339D</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation S371L</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation S373P</th>\n","      <td>0.99999997</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation S375F</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation K417N</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation N440K</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation G446S</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation S477N</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation T478K</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation E484A</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation Q493R</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation G496S</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation Q498R</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation N501Y</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation Y505H</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation T547K</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D614G</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation H655Y</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation N679K</th>\n","      <td>0.99999997</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation P681H</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation N764K</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D796Y</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation N856K</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation Q954H</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation N969K</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation L981F</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation V213G</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation S371F</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation T376A</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D405N</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation R408S</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation L452R</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation F486V</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation R158G</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation P681R</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D950N</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation A570D</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation T716I</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation S982A</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D1118H</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D80A</th>\n","      <td>0.99999997</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D215G</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation R246I</th>\n","      <td>0.99999998</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation E484K</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation A701V</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation D138Y</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation R190S</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation K417T</th>\n","      <td>0.99999999</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation T1027I</th>\n","      <td>1.00000000</td>\n","    </tr>\n","    <tr>\n","      <th>WT + mutation V1176F</th>\n","      <td>1.00000000</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                        tm_score\n","WT + mutation A67V    1.00000000\n","WT + mutation T95I    0.99999999\n","WT + mutation G142D   1.00000000\n","WT + mutation L212I   1.00000000\n","WT + mutation G339D   1.00000000\n","WT + mutation S371L   0.99999999\n","WT + mutation S373P   0.99999997\n","WT + mutation S375F   1.00000000\n","WT + mutation K417N   0.99999999\n","WT + mutation N440K   0.99999999\n","WT + mutation G446S   1.00000000\n","WT + mutation S477N   1.00000000\n","WT + mutation T478K   0.99999999\n","WT + mutation E484A   0.99999999\n","WT + mutation Q493R   1.00000000\n","WT + mutation G496S   1.00000000\n","WT + mutation Q498R   0.99999998\n","WT + mutation N501Y   0.99999999\n","WT + mutation Y505H   0.99999999\n","WT + mutation T547K   0.99999998\n","WT + mutation D614G   1.00000000\n","WT + mutation H655Y   0.99999999\n","WT + mutation N679K   0.99999997\n","WT + mutation P681H   1.00000000\n","WT + mutation N764K   0.99999999\n","WT + mutation D796Y   0.99999998\n","WT + mutation N856K   1.00000000\n","WT + mutation Q954H   0.99999999\n","WT + mutation N969K   0.99999999\n","WT + mutation L981F   0.99999999\n","WT + mutation V213G   0.99999999\n","WT + mutation S371F   0.99999998\n","WT + mutation T376A   0.99999999\n","WT + mutation D405N   0.99999999\n","WT + mutation R408S   0.99999999\n","WT + mutation L452R   1.00000000\n","WT + mutation F486V   1.00000000\n","WT + mutation R158G   1.00000000\n","WT + mutation P681R   1.00000000\n","WT + mutation D950N   1.00000000\n","WT + mutation A570D   1.00000000\n","WT + mutation T716I   0.99999999\n","WT + mutation S982A   0.99999998\n","WT + mutation D1118H  0.99999998\n","WT + mutation D80A    0.99999997\n","WT + mutation D215G   1.00000000\n","WT + mutation R246I   0.99999998\n","WT + mutation E484K   0.99999999\n","WT + mutation A701V   1.00000000\n","WT + mutation D138Y   0.99999999\n","WT + mutation R190S   1.00000000\n","WT + mutation K417T   0.99999999\n","WT + mutation T1027I  1.00000000\n","WT + mutation V1176F  1.00000000"]},"execution_count":69,"metadata":{},"output_type":"execute_result"}],"source":["import pandas as pd\n","pd.set_option(\"display.precision\", 8)\n","tm_scores = {key: value for key, value in tm_scores.items()}\n","df_tm = pd.DataFrame.from_dict(tm_scores, orient=\"index\", columns = [\"tm_score\"])\n","df_tm"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"yBNt4WsntWlJ"},"outputs":[],"source":["df_tm.to_csv(\"Data\\single_mutations_tm_scores.csv\")"]},{"cell_type":"code","source":["tm_scores_conf = {}\n","\n","for conformation in conformations:\n","\n","  labels = list(spike_muts_variants.keys())\n","  N = len(labels)\n","  tm_scores = np.zeros((N,N))\n","  pdbWTPath =  cwd+\"/PDBs/{}/XBB.1_{}.pdb\".format(conformation, conformation)\n","  pdbFilesPath = cwd+\"/PDBs/\"\n","\n","  for i, (variant1, mut1) in enumerate(spike_muts_variants.items()):\n","      for j, (variant2, mut2) in enumerate(spike_muts_variants.items()):\n","\n","          if i > j:\n","              if variant1!=variant2:\n","                  print(variant1, variant2)\n","\n","                  if len(mut1) == 0:\n","                      mut1 = None\n","                      pdbFilePath1 = pdbFilesPath+conformation+\"/XBB.1_{}.pdb\".format(conformation)\n","                  else:\n","                      pdbFilePath1 = pdbFilesPath+conformation+\"/{}_{}.pdb\".format(variant1, conformation)\n","\n","                  if len(mut2) == 0:\n","                      mut2 = None\n","                      pdbFilePath2 = pdbFilesPath+conformation+\"/XBB.1_{}.pdb\".format(conformation)\n","                  else:\n","                      pdbFilePath2 = pdbFilesPath+conformation+\"/{}_{}.pdb\".format(variant2, conformation)\n","\n","                  tm_score = compute_tm_score(pdbFilePath1, pdbFilePath2)*100\n","\n","              else:\n","                  tm_score = 100\n","\n","              print(\"{} vs {}. TM score: {}\".format(variant1, variant2, tm_score))\n","              tm_scores[i, j] = tm_score\n","              tm_scores[j, i] = tm_score\n","\n","          elif i == j:\n","              tm_scores[i, j] = 100\n","\n","  tm_scores_conf[conformation] = tm_scores"],"metadata":{"id":"jzK_H4AewIOG","executionInfo":{"status":"ok","timestamp":1693919499409,"user_tz":-60,"elapsed":57176,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"e305d9c5-dc45-4c34-83cb-0c4c4c0f2a2f"},"execution_count":33,"outputs":[{"output_type":"stream","name":"stdout","text":["XBB.1.9.1 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.9.1 vs XBB.1. TM score: 99.99999808039482\n","XBB.2.3 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.2.3 vs XBB.1. TM score: 99.99999593617336\n","XBB.2.3 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.2.3 vs XBB.1.9.1. TM score: 99.99999785912162\n","XBB.1.5 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.1. TM score: 99.99999808039482\n","XBB.1.5 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.1.9.1. TM score: 100.0\n","XBB.1.5 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.2.3. TM score: 99.99999785912162\n","XBB.1.16 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1. TM score: 99.99999784293418\n","XBB.1.16 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1.9.1. TM score: 99.9999997621694\n","XBB.1.16 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.2.3. TM score: 99.99999762132452\n","XBB.1.16 XBB.1.5\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1.5. TM score: 99.9999997621694\n","EG.5.1 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1. TM score: 99.99999689792294\n","EG.5.1 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.9.1. TM score: 99.9999988159321\n","EG.5.1 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.2.3. TM score: 99.9999966767162\n","EG.5.1 XBB.1.5\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.5. TM score: 99.9999988159321\n","EG.5.1 XBB.1.16\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.16. TM score: 99.99999857731355\n","XBB.1.9.1 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.9.1 vs XBB.1. TM score: 99.99999802107958\n","XBB.2.3 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.2.3 vs XBB.1. TM score: 99.99999566358758\n","XBB.2.3 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.2.3 vs XBB.1.9.1. TM score: 99.9999976464848\n","XBB.1.5 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.1. TM score: 99.99999802107958\n","XBB.1.5 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.1.9.1. TM score: 100.0\n","XBB.1.5 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.2.3. TM score: 99.9999976464848\n","XBB.1.16 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1. TM score: 99.99999768352417\n","XBB.1.16 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1.9.1. TM score: 99.99999966235832\n","XBB.1.16 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.2.3. TM score: 99.9999973084166\n","XBB.1.16 XBB.1.5\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1.5. TM score: 99.99999966235832\n","EG.5.1 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1. TM score: 99.99999662868979\n","EG.5.1 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.9.1. TM score: 99.99999860638017\n","EG.5.1 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.2.3. TM score: 99.9999962545144\n","EG.5.1 XBB.1.5\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.5. TM score: 99.99999860638017\n","EG.5.1 XBB.1.16\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.16. TM score: 99.99999826870906\n","XBB.1.9.1 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.9.1 vs XBB.1. TM score: 99.99999819865518\n","XBB.2.3 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.2.3 vs XBB.1. TM score: 99.99999605290851\n","XBB.2.3 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.2.3 vs XBB.1.9.1. TM score: 99.99999785768077\n","XBB.1.5 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.1. TM score: 99.99999819865518\n","XBB.1.5 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.1.9.1. TM score: 100.0\n","XBB.1.5 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.5 vs XBB.2.3. TM score: 99.99999785768077\n","XBB.1.16 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1. TM score: 99.99999789139032\n","XBB.1.16 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1.9.1. TM score: 99.9999996926551\n","XBB.1.16 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.2.3. TM score: 99.99999754997138\n","XBB.1.16 XBB.1.5\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["XBB.1.16 vs XBB.1.5. TM score: 99.9999996926551\n","EG.5.1 XBB.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1. TM score: 99.9999969311977\n","EG.5.1 XBB.1.9.1\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.9.1. TM score: 99.99999873148035\n","EG.5.1 XBB.2.3\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.2.3. TM score: 99.99999659058689\n","EG.5.1 XBB.1.5\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]},{"output_type":"stream","name":"stdout","text":["EG.5.1 vs XBB.1.5. TM score: 99.99999873148035\n","EG.5.1 XBB.1.16\n","EG.5.1 vs XBB.1.16. TM score: 99.99999842411184\n"]},{"output_type":"stream","name":"stderr","text":["<ipython-input-21-183869b24821>:4: DeprecationWarning: Using .default_errordef() is deprecated. Use .errordef instead\n","  alignment.optimise()\n"]}]},{"cell_type":"code","execution_count":34,"metadata":{"id":"Bj43yzs5tWlJ","executionInfo":{"status":"ok","timestamp":1693919499409,"user_tz":-60,"elapsed":9,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}}},"outputs":[],"source":["import pandas as pd\n","\n","dfs_conformations = {}\n","\n","for conformation in conformations:\n","  tm_scores = tm_scores_conf[conformation]\n","  pd.set_option(\"display.precision\", 8)\n","  df_tm_variants = pd.DataFrame(tm_scores, columns = labels, index = labels)\n","  dfs_conformations[conformation] = df_tm_variants\n","  df_tm_variants.to_csv(cwd+\"/Data/tm_score_variants_{}.csv\".format(conformation))"]},{"cell_type":"code","execution_count":35,"metadata":{"id":"QylpOyTStWlJ","executionInfo":{"status":"ok","timestamp":1693919499409,"user_tz":-60,"elapsed":8,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"colab":{"base_uri":"https://localhost:8080/","height":237},"outputId":"d0c8809b-4d05-41b3-f7df-b5e5f7e62b88"},"outputs":[{"output_type":"execute_result","data":{"text/plain":["                  XBB.1     XBB.1.9.1       XBB.2.3       XBB.1.5  \\\n","XBB.1      100.00000000   99.99999808   99.99999594   99.99999808   \n","XBB.1.9.1   99.99999808  100.00000000   99.99999786  100.00000000   \n","XBB.2.3     99.99999594   99.99999786  100.00000000   99.99999786   \n","XBB.1.5     99.99999808  100.00000000   99.99999786  100.00000000   \n","XBB.1.16    99.99999784   99.99999976   99.99999762   99.99999976   \n","EG.5.1      99.99999690   99.99999882   99.99999668   99.99999882   \n","\n","               XBB.1.16        EG.5.1  \n","XBB.1       99.99999784   99.99999690  \n","XBB.1.9.1   99.99999976   99.99999882  \n","XBB.2.3     99.99999762   99.99999668  \n","XBB.1.5     99.99999976   99.99999882  \n","XBB.1.16   100.00000000   99.99999858  \n","EG.5.1      99.99999858  100.00000000  "],"text/html":["\n","  <div id=\"df-78a7fb46-8b09-4e2d-8b88-c844803f4b48\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>XBB.1</th>\n","      <th>XBB.1.9.1</th>\n","      <th>XBB.2.3</th>\n","      <th>XBB.1.5</th>\n","      <th>XBB.1.16</th>\n","      <th>EG.5.1</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>XBB.1</th>\n","      <td>100.00000000</td>\n","      <td>99.99999808</td>\n","      <td>99.99999594</td>\n","      <td>99.99999808</td>\n","      <td>99.99999784</td>\n","      <td>99.99999690</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.9.1</th>\n","      <td>99.99999808</td>\n","      <td>100.00000000</td>\n","      <td>99.99999786</td>\n","      <td>100.00000000</td>\n","      <td>99.99999976</td>\n","      <td>99.99999882</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.2.3</th>\n","      <td>99.99999594</td>\n","      <td>99.99999786</td>\n","      <td>100.00000000</td>\n","      <td>99.99999786</td>\n","      <td>99.99999762</td>\n","      <td>99.99999668</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.5</th>\n","      <td>99.99999808</td>\n","      <td>100.00000000</td>\n","      <td>99.99999786</td>\n","      <td>100.00000000</td>\n","      <td>99.99999976</td>\n","      <td>99.99999882</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.16</th>\n","      <td>99.99999784</td>\n","      <td>99.99999976</td>\n","      <td>99.99999762</td>\n","      <td>99.99999976</td>\n","      <td>100.00000000</td>\n","      <td>99.99999858</td>\n","    </tr>\n","    <tr>\n","      <th>EG.5.1</th>\n","      <td>99.99999690</td>\n","      <td>99.99999882</td>\n","      <td>99.99999668</td>\n","      <td>99.99999882</td>\n","      <td>99.99999858</td>\n","      <td>100.00000000</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-78a7fb46-8b09-4e2d-8b88-c844803f4b48')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-78a7fb46-8b09-4e2d-8b88-c844803f4b48 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-78a7fb46-8b09-4e2d-8b88-c844803f4b48');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","<div id=\"df-319d3406-db51-4d36-8a65-f06df0983adf\">\n","  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-319d3406-db51-4d36-8a65-f06df0983adf')\"\n","            title=\"Suggest charts.\"\n","            style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","  </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","  <script>\n","    async function quickchart(key) {\n","      const quickchartButtonEl =\n","        document.querySelector('#' + key + ' button');\n","      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","      quickchartButtonEl.classList.add('colab-df-spinner');\n","      try {\n","        const charts = await google.colab.kernel.invokeFunction(\n","            'suggestCharts', [key], {});\n","      } catch (error) {\n","        console.error('Error during call to suggestCharts:', error);\n","      }\n","      quickchartButtonEl.classList.remove('colab-df-spinner');\n","      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","    }\n","    (() => {\n","      let quickchartButtonEl =\n","        document.querySelector('#df-319d3406-db51-4d36-8a65-f06df0983adf button');\n","      quickchartButtonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","    })();\n","  </script>\n","</div>\n","    </div>\n","  </div>\n"]},"metadata":{},"execution_count":35}],"source":["dfs_conformations[\"closed\"]"]},{"cell_type":"code","source":["dfs_conformations[\"open\"]"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":237},"id":"sMOn712Hdw_x","executionInfo":{"status":"ok","timestamp":1693919499410,"user_tz":-60,"elapsed":8,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"2601a918-26d6-450f-8762-8f9a8bcec1c6"},"execution_count":36,"outputs":[{"output_type":"execute_result","data":{"text/plain":["                  XBB.1     XBB.1.9.1       XBB.2.3       XBB.1.5  \\\n","XBB.1      100.00000000   99.99999802   99.99999566   99.99999802   \n","XBB.1.9.1   99.99999802  100.00000000   99.99999765  100.00000000   \n","XBB.2.3     99.99999566   99.99999765  100.00000000   99.99999765   \n","XBB.1.5     99.99999802  100.00000000   99.99999765  100.00000000   \n","XBB.1.16    99.99999768   99.99999966   99.99999731   99.99999966   \n","EG.5.1      99.99999663   99.99999861   99.99999625   99.99999861   \n","\n","               XBB.1.16        EG.5.1  \n","XBB.1       99.99999768   99.99999663  \n","XBB.1.9.1   99.99999966   99.99999861  \n","XBB.2.3     99.99999731   99.99999625  \n","XBB.1.5     99.99999966   99.99999861  \n","XBB.1.16   100.00000000   99.99999827  \n","EG.5.1      99.99999827  100.00000000  "],"text/html":["\n","  <div id=\"df-80cd6142-a50c-47f9-9b49-40c72659c55f\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>XBB.1</th>\n","      <th>XBB.1.9.1</th>\n","      <th>XBB.2.3</th>\n","      <th>XBB.1.5</th>\n","      <th>XBB.1.16</th>\n","      <th>EG.5.1</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>XBB.1</th>\n","      <td>100.00000000</td>\n","      <td>99.99999802</td>\n","      <td>99.99999566</td>\n","      <td>99.99999802</td>\n","      <td>99.99999768</td>\n","      <td>99.99999663</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.9.1</th>\n","      <td>99.99999802</td>\n","      <td>100.00000000</td>\n","      <td>99.99999765</td>\n","      <td>100.00000000</td>\n","      <td>99.99999966</td>\n","      <td>99.99999861</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.2.3</th>\n","      <td>99.99999566</td>\n","      <td>99.99999765</td>\n","      <td>100.00000000</td>\n","      <td>99.99999765</td>\n","      <td>99.99999731</td>\n","      <td>99.99999625</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.5</th>\n","      <td>99.99999802</td>\n","      <td>100.00000000</td>\n","      <td>99.99999765</td>\n","      <td>100.00000000</td>\n","      <td>99.99999966</td>\n","      <td>99.99999861</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.16</th>\n","      <td>99.99999768</td>\n","      <td>99.99999966</td>\n","      <td>99.99999731</td>\n","      <td>99.99999966</td>\n","      <td>100.00000000</td>\n","      <td>99.99999827</td>\n","    </tr>\n","    <tr>\n","      <th>EG.5.1</th>\n","      <td>99.99999663</td>\n","      <td>99.99999861</td>\n","      <td>99.99999625</td>\n","      <td>99.99999861</td>\n","      <td>99.99999827</td>\n","      <td>100.00000000</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-80cd6142-a50c-47f9-9b49-40c72659c55f')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-80cd6142-a50c-47f9-9b49-40c72659c55f button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-80cd6142-a50c-47f9-9b49-40c72659c55f');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","<div id=\"df-c150e57b-9d8e-4f85-bff9-a35e9e4c8123\">\n","  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c150e57b-9d8e-4f85-bff9-a35e9e4c8123')\"\n","            title=\"Suggest charts.\"\n","            style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","  </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","  <script>\n","    async function quickchart(key) {\n","      const quickchartButtonEl =\n","        document.querySelector('#' + key + ' button');\n","      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","      quickchartButtonEl.classList.add('colab-df-spinner');\n","      try {\n","        const charts = await google.colab.kernel.invokeFunction(\n","            'suggestCharts', [key], {});\n","      } catch (error) {\n","        console.error('Error during call to suggestCharts:', error);\n","      }\n","      quickchartButtonEl.classList.remove('colab-df-spinner');\n","      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","    }\n","    (() => {\n","      let quickchartButtonEl =\n","        document.querySelector('#df-c150e57b-9d8e-4f85-bff9-a35e9e4c8123 button');\n","      quickchartButtonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","    })();\n","  </script>\n","</div>\n","    </div>\n","  </div>\n"]},"metadata":{},"execution_count":36}]},{"cell_type":"code","source":["dfs_conformations[\"complex\"]"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":237},"id":"UvKxOcdCdrAr","executionInfo":{"status":"ok","timestamp":1693919499410,"user_tz":-60,"elapsed":7,"user":{"displayName":"UGO LOMOIO","userId":"13693923129053979509"}},"outputId":"edfbb4a8-948a-47e6-f96b-acfcd3175d8a"},"execution_count":37,"outputs":[{"output_type":"execute_result","data":{"text/plain":["                  XBB.1     XBB.1.9.1       XBB.2.3       XBB.1.5  \\\n","XBB.1      100.00000000   99.99999820   99.99999605   99.99999820   \n","XBB.1.9.1   99.99999820  100.00000000   99.99999786  100.00000000   \n","XBB.2.3     99.99999605   99.99999786  100.00000000   99.99999786   \n","XBB.1.5     99.99999820  100.00000000   99.99999786  100.00000000   \n","XBB.1.16    99.99999789   99.99999969   99.99999755   99.99999969   \n","EG.5.1      99.99999693   99.99999873   99.99999659   99.99999873   \n","\n","               XBB.1.16        EG.5.1  \n","XBB.1       99.99999789   99.99999693  \n","XBB.1.9.1   99.99999969   99.99999873  \n","XBB.2.3     99.99999755   99.99999659  \n","XBB.1.5     99.99999969   99.99999873  \n","XBB.1.16   100.00000000   99.99999842  \n","EG.5.1      99.99999842  100.00000000  "],"text/html":["\n","  <div id=\"df-c45db5f9-9df4-4f8b-be4b-1db6c17345fd\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>XBB.1</th>\n","      <th>XBB.1.9.1</th>\n","      <th>XBB.2.3</th>\n","      <th>XBB.1.5</th>\n","      <th>XBB.1.16</th>\n","      <th>EG.5.1</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>XBB.1</th>\n","      <td>100.00000000</td>\n","      <td>99.99999820</td>\n","      <td>99.99999605</td>\n","      <td>99.99999820</td>\n","      <td>99.99999789</td>\n","      <td>99.99999693</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.9.1</th>\n","      <td>99.99999820</td>\n","      <td>100.00000000</td>\n","      <td>99.99999786</td>\n","      <td>100.00000000</td>\n","      <td>99.99999969</td>\n","      <td>99.99999873</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.2.3</th>\n","      <td>99.99999605</td>\n","      <td>99.99999786</td>\n","      <td>100.00000000</td>\n","      <td>99.99999786</td>\n","      <td>99.99999755</td>\n","      <td>99.99999659</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.5</th>\n","      <td>99.99999820</td>\n","      <td>100.00000000</td>\n","      <td>99.99999786</td>\n","      <td>100.00000000</td>\n","      <td>99.99999969</td>\n","      <td>99.99999873</td>\n","    </tr>\n","    <tr>\n","      <th>XBB.1.16</th>\n","      <td>99.99999789</td>\n","      <td>99.99999969</td>\n","      <td>99.99999755</td>\n","      <td>99.99999969</td>\n","      <td>100.00000000</td>\n","      <td>99.99999842</td>\n","    </tr>\n","    <tr>\n","      <th>EG.5.1</th>\n","      <td>99.99999693</td>\n","      <td>99.99999873</td>\n","      <td>99.99999659</td>\n","      <td>99.99999873</td>\n","      <td>99.99999842</td>\n","      <td>100.00000000</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-c45db5f9-9df4-4f8b-be4b-1db6c17345fd')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-c45db5f9-9df4-4f8b-be4b-1db6c17345fd button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-c45db5f9-9df4-4f8b-be4b-1db6c17345fd');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","<div id=\"df-ae308ef6-090f-4646-80dd-d39c89df8dcc\">\n","  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ae308ef6-090f-4646-80dd-d39c89df8dcc')\"\n","            title=\"Suggest charts.\"\n","            style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","  </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","  <script>\n","    async function quickchart(key) {\n","      const quickchartButtonEl =\n","        document.querySelector('#' + key + ' button');\n","      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","      quickchartButtonEl.classList.add('colab-df-spinner');\n","      try {\n","        const charts = await google.colab.kernel.invokeFunction(\n","            'suggestCharts', [key], {});\n","      } catch (error) {\n","        console.error('Error during call to suggestCharts:', error);\n","      }\n","      quickchartButtonEl.classList.remove('colab-df-spinner');\n","      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","    }\n","    (() => {\n","      let quickchartButtonEl =\n","        document.querySelector('#df-ae308ef6-090f-4646-80dd-d39c89df8dcc button');\n","      quickchartButtonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","    })();\n","  </script>\n","</div>\n","    </div>\n","  </div>\n"]},"metadata":{},"execution_count":37}]},{"cell_type":"code","source":[],"metadata":{"id":"erBClx4jdrDi"},"execution_count":null,"outputs":[]}],"metadata":{"colab":{"provenance":[]},"kernelspec":{"display_name":"PCN","language":"python","name":"pcn"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.3"}},"nbformat":4,"nbformat_minor":0}